components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -1612,4 +1613,27 @@ protected void unregisterConfigurationManager(ConfigurationManager configuration
         }
         OAuth2ServiceComponentHolder.getInstance().setConfigurationManager(null);
     }
+
+    @Reference(
+            name = "org.wso2.carbon.identity.handler.event.account.lock.service.AccountLockService",
+            service = AccountLockService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetAccountLockService"
+    )
+    protected void setAccountLockService(AccountLockService accountLockService) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("AccountLockService set in OAuth2ServiceComponent bundle.");
+        }
+        OAuth2ServiceComponentHolder.setAccountLockService(accountLockService);
+    }
+
+    protected void unsetAccountLockService(AccountLockService accountLockService) {
+
+        if (log.isDebugEnabled()) {
```

Comment:
same as the previous method.

Reply:
https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2581/commits/00551b73c2a1d5225a216da2da116d0368bd9479